### DISCUSSION TITLE
[QUESTION] compare_key in nested array in JSON

### DISCUSSION THREAD

**Discussion by 0123hoang** [jertel/elastalert2#850] [Q&A] [answered]
ELK version  7.13.2, Ubuntu 20.04

- My sample message in elk
```
{
   "cameras":[
      {
         "camID":"1",
         "msg":"{  \"code\" : 0,  \"description\" : \"Success\",  \"response\" : {    \"type\" : \"eos\",    \"source\" : \"1\",    \"timestamp\" : \"99:99:99.999999999\",    \"seqnum\" : 16667}}",
         "state":"-1"
      },
      {
         "camID":"2",
         "msg":"",
         "state":"0"
      }
   ],
   "metadata":{
      "BoxID":"Hi123"
   }
}
```
 - rule file
```
type: blacklist
compare_key:
 - "cameras.state" //Cannot set to cameras[0]state or cameras[1]state
blacklist:
 - "-1"
```
 - func compare (ruletypes.py:126)
```
class BlacklistRule(CompareRule):
    def compare(self, event):
        term = lookup_es_key(event, self.rules['compare_key'])
        if term in self.rules['blacklist']:
            return True
        return False
```
 - I see lookup_es_key() get "compare_key" env but return none.
 - If I change `self.rules['compare_key']` to 'cameras[0]state' it can get right value, but it cannot setting that value in rule file
 (ERROR:elastalert:Error running query: ['Failed to parse query [cameras[0]state:"-1"]'])
 - Did I do something wrong.

**Comment by jertel** [comment, marked as answer]
This appears to be a known issue. See https://github.com/Yelp/elastalert/issues/1570 and the mentioned workaround.
